lib/block_view.go

```diff
@@ -4726,135 +5590,780 @@ func (bav *UtxoView) _connectUpdateProfile(
+	} else if txMeta.SerialNumber > postEntry.NumNFTCopies {
+		return 0, 0, nil, RuleErrorNFTBidOnInvalidSerialNumber
+	}
 
-		// The StakeEntry is always left unmodified here.
+	// Validate the nftEntry.  Note that there is a special case where a bidder can submit a bid
+	// on SerialNumber zero.  This acts as a blanket bid on any serial number version of this NFT
+	// As a result, the nftEntry will be nil and should not be validated.
+	nftKey := MakeNFTKey(txMeta.NFTPostHash, txMeta.SerialNumber)
+	nftEntry := bav.GetNFTEntryForNFTKey(&nftKey)
+	bidderPKID := bav.GetPKIDForPublicKey(txn.PublicKey)
+	if bidderPKID == nil || bidderPKID.isDeleted {
+		return 0, 0, nil, fmt.Errorf("_connectNFTBid: PKID for bidder public key %v doesn't exist; this should never happen", string(txn.PublicKey))
+	}
 
-	} else {
-		// When there's no pre-existing profile entry we need to do more
-		// checks.
-		if len(txMeta.NewUsername) == 0 {
-			return 0, 0, nil, RuleErrorProfileUsernameTooShort
+	// Save a copy of the bid entry so that we can use it in the disconnect.
+	nftBidKey := MakeNFTBidKey(bidderPKID.PKID, txMeta.NFTPostHash, txMeta.SerialNumber)
+	prevNFTBidEntry := bav.GetNFTBidEntryForNFTBidKey(&nftBidKey)
+
+	if txMeta.SerialNumber != uint64(0) {
+		// Verify the NFT entry that is being bid on exists.
+		if nftEntry == nil || nftEntry.isDeleted {
+			return 0, 0, nil, RuleErrorNFTBidOnNonExistentNFTEntry
 		}
-		// We allow users to create profiles without a description or picture
-		// in the consensus code. If desired, frontends can filter out profiles
-		// that don't have these fields.
-		//
-		// Creator percentage and stake multiple are sufficiently checked above.
 
-		// In this case we need to set all the fields using what was passed
-		// into the transaction.
+		// Verify the NFT entry being bid on is for sale.
+		if !nftEntry.IsForSale {
+			return 0, 0, nil, RuleErrorNFTBidOnNFTThatIsNotForSale
+		}
 
-		// If below block height, user transaction public key.
-		// If above block height, use ProfilePublicKey if available.
-		profileEntryPublicKey := txn.PublicKey
-		if blockHeight > ParamUpdaterProfileUpdateFixBlockHeight {
-			profileEntryPublicKey = profilePublicKey
+		// Verify that the bidder is not the current owner of the NFT.
+		if reflect.DeepEqual(nftEntry.OwnerPKID, bidderPKID.PKID) {
+			return 0, 0, nil, RuleErrorNFTOwnerCannotBidOnOwnedNFT
 		}
 
-		newProfileEntry = ProfileEntry{
-			PublicKey:   profileEntryPublicKey,
-			Username:    txMeta.NewUsername,
-			Description: txMeta.NewDescription,
-			ProfilePic:  txMeta.NewProfilePic,
+		// Verify that the bid amount is greater than the min bid amount for this NFT.
+		// We allow BidAmountNanos to be 0 if there exists a previous bid entry. A value of 0 indicates that we should delete the entry.
+		if txMeta.BidAmountNanos < nftEntry.MinBidAmountNanos && !(txMeta.BidAmountNanos == 0 && prevNFTBidEntry != nil) {
+			return 0, 0, nil, RuleErrorNFTBidLessThanMinBidAmountNanos
+		}
```

Comment:
I'm confused-- why are all these checks wrapped in the if statement? Seems like they should be done even if it's a standing offer.